equinix/resource_fabric_cloud_router.go

```diff
@@ -15,47 +18,249 @@ import (
 
 	v4 "github.com/equinix-labs/fabric-go/fabric/v4"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func resourceCloudRouter() *schema.Resource {
+func FabricCloudRouterPackageSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"code": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Fabric Cloud Router package code",
+		},
+	}
+}
+func FabricCloudRouterAccountSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"account_number": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Optional:    true,
+			Description: "Account Number",
+		},
+	}
+}
+func FabricCloudRouterProjectSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"project_id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Optional:    true,
+			Description: "Project Id",
+		},
+		"href": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "Unique Resource URL",
+		},
+	}
+}
+
+func FabricCloudRouterResourceSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "Equinix-assigned Fabric Cloud Router identifier",
+		},
+		"href": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "Fabric Cloud Router URI information",
+		},
+		"name": {
```

Comment:
Good to put a validateFunc here to do a character count. Otherwise the user will hit an API error that will be harder to read then a validation check.